controllers/account-setting/manage-account/addTag.controller.js

```diff
@@ -1,19 +1,36 @@
 'use strict';
 
 const tagModel = require('../../models/tags.model');
-
+const UserModel = require('../../models/user.model'); 
 const addTagController = async (req, res, next) => {
 	try {
 		let { name, colorHex } = req.body;
 		const { id } = req.params;
 
-		//check tag existence
+		// Check user role
+		const user = await UserModel.findById(id);
+		if (!user) {
+			return res.status(404).send({
+				code: 404,
+				message: 'User not found'
+			});
+		}
+
+		// Restrict tag creation based on user role
+		if (user.role !== 'CRM_ADMIN') {
```

Comment:
This is redundant and bad practise. use a middleware. Already implemented with routes in notifications or auth